playwright/masonry/scroll-performance.spec.mjs

```diff
@@ -2,67 +2,122 @@
 import fs from 'fs';
 import { expect, test } from '@playwright/test';
 import getServerURL from './utils/getServerURL.mjs';
-import {
-  eventsToCsv,
-  getFpsFromEvents,
-  getFpsMetricsPerScroll,
-  metricsToCsv,
-} from './utils/tracingEvents.mjs';
 import waitForRenderedItems from './utils/waitForRenderedItems.mjs';
 
+const DEFAULT_SCROLL_COUNT = 5;
+const PINS_COUNT = 20;
+const PAGE_HEIGHT = 1000;
+
 /*::
 type Event = {
   name: string,
   ts: number,
-  fps?: number,
+  args?: { frameSeqId?: number },
 };
 */
 
-const DEFAULT_SCROLL_COUNT = 5;
-const PINS_COUNT = 20;
-const PAGE_HEIGHT = 1000;
+function isValidEvent(event /*: Event */) {
+  return event.name === 'clock_sync' || event.args?.frameSeqId;
+}
+
+function microsecondsToMilliseconds(microseconds /*: number */) {
+  return microseconds / 1e3;
+}
+
+/*
+  Parse out the following frame data from the collected trace events:
+  - droppedFrames: These are events with the name DroppedFrame and represent frames that were either dropped or not fully rendered
+  - completeFrames: These are events with the name DrawFrame and represent frames that were fully rendered
+  - frameDuration: This is the duration of each frame, either complete or dropped. A frame duration is calculated as the time between the DrawFrame or DroppedFrame event and the next BeginFrame event
+*/
+function parseFrameData(events /*: $ReadOnlyArray<Event> */) {
+  let currentFrameStartTime;
+  const droppedFrames = events.filter((e) => e.name === 'DroppedFrame').length;
+  const completeFrames = events.filter((e) => e.name === 'DrawFrame').length;
+  const frameDuration = events.reduce(
+    (acc, event) => {
+      if (event.name === 'BeginFrame') {
+        if (currentFrameStartTime != null) {
+          // ts is stored in microseconds. See: https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/edit
+          const delta = microsecondsToMilliseconds(
+            event.ts - currentFrameStartTime
+          );
+          acc.push(delta);
+        }
+      }
+      if (event.name === 'DrawFrame' || event.name === 'DroppedFrame') {
+        currentFrameStartTime = event.ts;
```

Comment:
good question - this doesn't work because it doesn't account for the duration of `DrawFrame`. in order to do that, we'd have to calculate the delta between `DrawFrame`  -> `BeginFrame` (kind of silly imo)